DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -71,32 +89,92 @@ public IActionResult EditRegistrationPrompt(EditRegistrationPromptViewModel mode
             {
                 SaveAction => EditRegistrationPromptPostSave(model),
                 AddPromptAction => RegistrationPromptAnswersPostAddPrompt(model),
+                BulkAction => EditRegistrationPromptBulk(model),
                 _ => RedirectToAction("Error", "LearningSolutions")
             };
         }
 
         [HttpGet]
-        [Route("Add/SelectPrompt")]
-        public IActionResult AddRegistrationPromptSelectPrompt()
+        [Route("{promptNumber:int}/Edit/Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditRegistrationPromptData>))]
+        public IActionResult EditRegistrationPromptBulk(int promptNumber)
         {
-            var addRegistrationPromptData = TempData.Peek<AddRegistrationPromptData>();
+            var data = TempData.Peek<EditRegistrationPromptData>()!;
 
-            if (addRegistrationPromptData == null || !Request.Cookies.ContainsKey(CookieName))
+            var model = new BulkRegistrationPromptAnswersViewModel(
+                data.EditModel.OptionsString,
+                false,
+                promptNumber
+            );
+
+            return View("BulkRegistrationPromptAnswers", model);
+        }
+
+        [HttpPost]
+        [Route("Edit/Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionDataXor<AddRegistrationPromptData, EditRegistrationPromptData>))]

```

Comment:
Do we still need the Xor service filter?

Reply:
Ah no, probably not anymore